src/components/observer/mutation_observer/mutation_observer.stories.tsx

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useState } from 'react';
+import type { Meta, StoryObj } from '@storybook/react';
+
+import { _EuiButtonColor } from '../../../themes/amsterdam/global_styling/mixins';
+import { EuiFlexGroup, EuiFlexItem } from '../../flex';
+import { EuiSpacer } from '../../spacer';
+import { EuiButton, EuiButtonEmpty } from '../../button';
+import { EuiPanel } from '../../panel';
+import {
+  EuiMutationObserver,
+  EuiMutationObserverProps,
+} from './mutation_observer';
+
+const meta: Meta<EuiMutationObserverProps> = {
+  title: 'Utilities/EuiMutationObserver',
+  component: EuiMutationObserver,
+};
+
+export default meta;
+type Story = StoryObj<EuiMutationObserverProps>;
+
+const StatefulPlayground = (
+  props: Omit<EuiMutationObserverProps, 'onMutation'>
+) => {
+  const [lastMutation, setLastMutation] = useState('no changes detected');
+  const [buttonColor, setButtonColor] = useState<_EuiButtonColor>('primary');
+  const [items, setItems] = useState(['Item 1', 'Item 2', 'Item 3']);
+
+  const toggleButtonColor = () => {
+    setButtonColor(buttonColor === 'primary' ? 'warning' : 'primary');
+  };
+
+  const addItem = () => {
+    setItems([...items, `Item ${items.length + 1}`]);
+  };
+
+  const handleOnMutation = ([{ type }]: MutationRecord[]) => {
+    setLastMutation(
+      type === 'attributes' ? 'button class name changed' : 'DOM tree changed'
+    );
+  };
+
+  return (
+    <>
+      <p>{lastMutation}</p>
+
+      <EuiSpacer />
+
+      <EuiMutationObserver
+        observerOptions={{ subtree: true, attributes: true, childList: true }}
+        onMutation={handleOnMutation}
+        {...props}
```

Comment:
This isn't working because of the `onMutation` handler Storybook is automatically passing to the component:

<img width="460" alt="" src="https://github.com/elastic/eui/assets/549407/2fc64a57-6947-46fc-b568-822077d24fea">

We need to either:

1. Remove `lastMutation` / `handleOnMutation` and just let the `onMutation` action tell consumers what's happening (not my personal preference)

or

2. Change this to
    ```suggestion
    {...props}
    onMutation={handleOnMutation}
   ```
   and then update `handleOnMutation` to call a Storybook action handler automatically. If you search for `@storybook/addon-actions'` through the EUI codebase you can see a [couple examples](https://github.com/elastic/eui/blob/75d5b5afe7179dbc2764dd48078a8712b356366c/src/components/combo_box/combo_box.stories.tsx#L69-L72) of this, but a quick example would be:
    ```ts
    import { action } from '@storybook/addon-actions';

    const handleOnMutation = (mutationRecords: MutationRecord[], ...args) => {
      const [{ type }] = mutationRecords;
      // ... default handler
      action('onMutation')(mutationRecords, ...args);
    };
    ```

Reply:
Ooh or actually, the way you've done it in the resize observer story might be even better/not require the import:

```tsx
const StatefulPlayground = (
  { onMutation, ...props }: EuiMutationObserverProps
) => {

const handleOnMutation = (mutationRecords: MutationRecord[], ...args) => {
  const [{ type }] = mutationRecords;
  // ... default handler
  onMutation(mutationRecords, ...args);
};
```

Reply:
Ah yeah, that's my bad! I had it the same way and it was working but somehow I decided to remove the call to `onMutation` again 🙈 Thanks for the catch!